exonum-java-binding/core/src/main/java/com/exonum/binding/core/blockchain/Blockchain.java

```diff
@@ -140,23 +151,52 @@ public long getHeight() {
   }
 
   /**
-   * Returns a map with a key-value pair of a transaction hash and execution result. Note that this
-   * is a <a href="ProofMapIndexProxy.html#key-hashing">proof map that uses non-hashed keys</a>.
+   * Returns execution errors that occurred in the block at the given height. Execution errors
+   * are preserved for transactions and before/after transaction handlers.
+   *
+   * <p>The {@linkplain ProofMapIndexProxy#getIndexHash() index hash} of this index is recorded
+   * in the block header as <!-- todo: link (ECR-4021) --> {@code Block#getErrorHash()}. That
+   * enables constructing proofs
+   * <!-- todo: link 'proofs' (where do we document construction procedure?) -->
+   * that a transaction with a certain message hash was executed at a certain
+   * <em>{@linkplain TransactionLocation location}</em> with a particular result.
+   *
+   * @param blockHeight the height of the block
+   * @throws IllegalArgumentException if the height is invalid: negative or exceeding
+   *     the {@linkplain #getHeight() blockchain height}
    */
-  public ProofMapIndexProxy<HashCode, ExecutionStatus> getTxResults() {
-    return schema.getTxResults();
+  public ProofMapIndexProxy<CallInBlock, ExecutionError> getCallErrors(long blockHeight) {
+    return schema.getCallErrors(blockHeight);
   }
 
   /**
-   * Returns a transaction execution result for given message hash.
+   * Returns a transaction execution result for the given message hash.
    *
    * @return a transaction execution result, or {@code Optional.empty()} if this transaction
    *         is unknown or was not yet executed
    */
   public Optional<ExecutionStatus> getTxResult(HashCode messageHash) {
-    MapIndex<HashCode, ExecutionStatus> txResults = getTxResults();
-    ExecutionStatus transactionResult = txResults.get(messageHash);
-    return Optional.ofNullable(transactionResult);
+    // Find tx location
+    Optional<TransactionLocation> txLocationOpt = getTxLocation(messageHash);
+    if (txLocationOpt.isPresent()) {
```

Comment:
Nit:
How about to slightly refactor this code:
1) Extract method `ExecutionStatus getExecutionStatus(TransactionLocation txLocation)`
2) Use declarative way:
```
return getTxLocation(messageHash)
.map(this::getExecutionStatus);
```

Reply:
Refactored.